client/remoterepo.c

```diff
@@ -271,18 +320,55 @@ TDNFCheckHash(
 uint32_t
 TDNFCheckRepoMDFileHashFromMetalink(
     char *pszFile,
-    TDNF_METALINK_FILE *ml_file
+    TDNF_ML_CTX *ml_ctx
     )
 {
     uint32_t dwError = 0;
+    TDNF_ML_HASH_LIST *hashList = NULL;
+    TDNF_ML_HASH_INFO *hashInfo = NULL;
+    unsigned char digest[EVP_MAX_MD_SIZE] = {0};
+    int hashType = -1;
+    TDNF_ML_HASH_INFO *currHashInfo = NULL;
+    int currHashType = TDNF_HASH_SENTINEL;
+
     if(IsNullOrEmptyString(pszFile) ||
-       !ml_file)
+       !ml_ctx)
     {
         dwError = ERROR_TDNF_INVALID_PARAMETER;
         BAIL_ON_TDNF_ERROR(dwError);
     }
 
-    dwError = TDNFCheckHash(pszFile, ml_file->digest, ml_file->type);
+    hashList = ml_ctx->hashes;
+    while(hashList)
+    {
+        currHashInfo = NULL;
```

Comment:
Remove this line, i see we are initialising it twice.

Reply:
done